lua/telescope/builtin/init.lua

```diff
@@ -64,7 +64,13 @@ builtin.spell_suggest = require('telescope.builtin.internal').spell_suggest
 builtin.lsp_references = require('telescope.builtin.lsp').references
 builtin.lsp_document_symbols = require('telescope.builtin.lsp').document_symbols
 builtin.lsp_code_actions = require('telescope.builtin.lsp').code_actions
-builtin.lsp_diagnostics = require('telescope.builtin.lsp').diagnostics
+builtin.lsp_buffer_diagnostics = require('telescope.builtin.lsp').diagnostics
+builtin.lsp_workspace_diagnostics = function(opts)
+  opts = vim.F.if_nil(opts, {})
+  opts.hide_filename = false
+  opts.get_all = true
+  require('telescope.builtin.lsp').diagnostics(opts)
+end
```

Comment:
Not the biggest fan of code in this file. Move it to lsp and if you have problems with the capabilities check you can use sunjons `feature_map` to go from `document_diagnostics` -> `diagnostics` and from `workspace_diagnostics` -> `diagnostics`
Thanks :)

Reply:
On second thought (first being this commit x) I fully agree!

`capabilities_check` is omitted since diagnostics are not there captured (cf. screenshot):

![sumneko_capabilities](https://user-images.githubusercontent.com/39233597/109954148-43af9b80-7ce1-11eb-8817-33336d39cfce.png)

As per [upstream implementation](https://github.com/neovim/neovim/blob/df4440024bb1f1ce368f5e5844d8af925e264b63/runtime/lua/vim/lsp/diagnostic.lua#L330) an empty table should be returned which ultimately triggers `print('No diagnostics found')`.

Would you want this message to be more explicit in such a case, i.e. in case the server doesn't support diagnostics at all?

Reply:
Oh i just assumed it would be in capabilities :laughing: . I think message is good enough.